cpp/include/cudf/strings/convert/convert_durations.hpp

```diff
@@ -0,0 +1,128 @@
+ * | :-------: | ----------- | ---------------- |
+ * | %%%% | A literal % character | '%' |
+ * | %%n | A newline character | '\n' |
+ * | %%t | A horizontal tab character | '\t' |
+ * | %%D | Days | -2,147,483,648 to 2,147,483,647 |
+ * | %%H | 24-hour of the day | 00 to 23 |
+ * | %%I | 12-hour of the day | 00 to 11 |
+ * | %%M | Minute of the hour | 00 to 59 |
+ * | %%S | Second of the minute | 00 to 59.999999999 |
+ * | %%p | AM/PM designations associated with a 12-hour clock | 'AM' or 'PM' |
+ * | %%R | Equivalent to "%H:%M" |  |
+ * | %%T | Equivalent to "%H:%M:%S" |  |
+ * | %%r | The locale's 12-hour clock time (without sign) |  |
+ * | %%OH | locale's alternative representation of 24-hour of the day (without sign) | 00 to 23 |
+ * | %%OI | locale's alternative representation of 12-hour of the day (without sign) | 00 to 11 |
+ * | %%OM | locale's alternative representation of Minute of the hour (without sign) | 00 to 59 |
+ * | %%OS | locale's alternative representation of Second of the minute (without sign) | 00 to 59 |
+ *
+ * Other specifiers are not currently supported.
+ *
+ * Invalid formats are not checked. If the string contains unexpected
+ * or insufficient characters, that output row entry's duration value is undefined.
+ *
+ * Any null string entry will result in a corresponding null row in the output column.
+ *
+ * The resulting time units are specified by the `duration_type` parameter.
+ *
+ * @throw cudf::logic_error if duration_type is not a duration type.
+ *
+ * @param strings Strings instance for this operation.
+ * @param duration_type The duration type used for creating the output column.
+ * @param format String specifying the duration format in strings.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New duration column.
+ */
+std::unique_ptr<column> to_durations(
+  strings_column_view const& strings,
+  data_type duration_type,
+  std::string const& format,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+/**
+ * @brief Returns a new strings column converting a duration column into
+ * strings using the provided format pattern.
+ *
+ * The format pattern can include the following specifiers:
+ * "%%,%n,%t,%D,%H,%I,%M,%S,%p,%R,%T,%r,%OH,%OI,%OM,%OS"
+ *
+ * | Specifier | Description | Range |
+ * | :-------: | ----------- | ---------------- |
+ * | %%%% | A literal % character | '%' |
+ * | %%n | A newline character | '\n' |
+ * | %%t | A horizontal tab character | '\t' |
+ * | %%D | Days | -2,147,483,648 to 2,147,483,647 |
+ * | %%H | 24-hour of the day | 00 to 23 |
+ * | %%I | 12-hour of the day | 00 to 11 |
+ * | %%M | Minute of the hour | 00 to 59 |
+ * | %%S | Second of the minute | 00 to 59.999999999 |
+ * | %%p | AM/PM designations associated with a 12-hour clock | 'AM' or 'PM' |
+ * | %%R | Equivalent to "%H:%M" |  |
+ * | %%T | Equivalent to "%H:%M:%S" |  |
+ * | %%r | The locale's 12-hour clock time (without sign) |  |
```

Comment:
I'm uncomfortable using the word _locale_ here. It implies that we are sensitive to the running OS locale settings which the code does not support. For reference see the `%O` specifiers here: https://en.cppreference.com/w/cpp/chrono/c/strftime
The `%O` implies different results for the Japanese locale.

I would recommend dropping _locale_ from the descriptions here and just include what each of these actually does:
```
%r  = Equivalent to "%I:%M:%S %p"
%OH = same as %H but without sign
%OI = same as %I but without sign
%OM = same as %M but without sign
%OS = same %S but without sign
```

Reply:
updated.